src-newer-jdks/orchard/java/parser_next.clj

```diff
@@ -323,14 +323,14 @@
         (when-let [^DocletEnvironment root (parse-java path (module-name klass))]
           (try
             (let [path-resource (io/resource path)]
-              (assoc (->> (.getIncludedElements root)
-                          (filterv #(#{ElementKind/CLASS
-                                       ElementKind/INTERFACE
-                                       ElementKind/ENUM}
-                                     (.getKind ^Element %)))
-                          (mapv #(parse-info % root))
-                          (filterv #(= klass (:class %)))
-                          (first))
```

Comment:
Because of eagerness, this code invoked `parse-info` on every element of `root` even though we only need the first one that matches `(= klass (:class %)`.